huaweicloud/services/acceptance/vpc/resource_huaweicloud_vpc_sub_network_interface_test.go

```diff
@@ -0,0 +1,149 @@
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving sub network interface: %s", err)
+	}
+
+	return utils.FlattenResponse(getSubNetworkInterfaceResp)
+}
+
+func TestAccSubNetworkInterface_basic(t *testing.T) {
+	var (
+		sub_network_interface interface{}
+		name                  = acceptance.RandomAccResourceName()
+		rName                 = "huaweicloud_vpc_sub_network_interface.test"
+		rc                    = acceptance.InitResourceCheck(
+			rName,
+			&sub_network_interface,
+			getSubNetworkInterfaceResourceFunc,
+		)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testSubNetworkInterface_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "subnet_id", "huaweicloud_vpc_subnet.test", "id"),
+					resource.TestCheckResourceAttrPair(rName, "parent_id", "huaweicloud_compute_instance.test", "network.0.port"),
+					resource.TestCheckResourceAttr(rName, "description", "created by test acc"),
+					resource.TestCheckResourceAttr(rName, "vlan_id", "3002"),
+					resource.TestCheckResourceAttr(rName, "security_group_ids.#", "1"),
+					resource.TestCheckResourceAttrSet(rName, "ip_address"),
+					resource.TestCheckResourceAttrSet(rName, "mac_address"),
+					resource.TestCheckResourceAttrSet(rName, "parent_device_id"),
+					resource.TestCheckResourceAttrSet(rName, "vpc_id"),
+					resource.TestCheckResourceAttrSet(rName, "status"),
+					resource.TestCheckResourceAttrSet(rName, "created_at"),
+				),
+			},
+			{
+				Config: testSubNetworkInterface_basic_update(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "subnet_id", "huaweicloud_vpc_subnet.test", "id"),
+					resource.TestCheckResourceAttrPair(rName, "parent_id", "huaweicloud_compute_instance.test", "network.0.port"),
+					resource.TestCheckResourceAttr(rName, "description", ""),
+					resource.TestCheckResourceAttr(rName, "vlan_id", "3002"),
+					resource.TestCheckResourceAttr(rName, "security_group_ids.#", "1"),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testaccSubNetworkInterface_base(name string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+data "huaweicloud_availability_zones" "test" {}
+
+data "huaweicloud_images_images" "test" {
+  name        = "Ubuntu 22.04 server 64bit"
+  visibility  = "public"
+}
+
+resource "huaweicloud_compute_instance" "test" {
+  name               = "%[2]s"
+  image_id           = data.huaweicloud_images_images.test.images[0].id
+  flavor_id          = "c7.large.2"
```

Comment:
hard coding，please using data source to query the result, c7 flavor means follow configuration:
- genaration: c7
- performance_type: computingv3

Reply:
done